### ISSUE TITLE
arr.h hava a bug

### ISSUE THREAD

**Issue by zhangjg** [YasserAsmi/jvar#8]
in the file of arr.h  at line 1084 :
```c++
   StrArray() :   ObjArray(StrArray::compare)
```
in centos 6 ,have error, like:
class ‘jvar::StrArray’ does not have any field named ‘ObjArray’
maybe shold change to
```c++
   StrArray(): ObjArray<std::string>(StrArray::compare)
```

**Comment by aelzenaar** [comment]
Which compiler are you using?

**Comment by zhangjg** [comment]
g++

**Comment by aelzenaar** [comment]
This is a compiler bug fixed in G++ 4.5 (the version shipping with CentOS 6 is 4.4, I believe). See the first answer to [this stack overflow question](http://stackoverflow.com/questions/8887864/template-base-constructor-call-in-member-initialization-list-error).

Your solution is valid C++ and I'll submit a pull request, but you might want to upgrade your compiler anyway (if possible) because it is quite old!

Are there any other classes that are giving out errors once the one above is fixed?

**Comment by YasserAsmi** [comment]
Merged aelzenaar's fix.  Thanks!
